editor/editor_audio_buses.cpp

```diff
@@ -245,7 +245,17 @@ void EditorAudioBus::update_bus() {
 		fx->set_cell_mode(0, TreeItem::CELL_MODE_CHECK);
 		fx->set_editable(0, true);
 		fx->set_checked(0, AudioServer::get_singleton()->is_bus_effect_enabled(index, i));
-		fx->set_text(0, afx->get_name());
+
+		if (afx->get_name().is_empty()) { // if the effect has no name use the file name
+			fx->set_text(0, afx->get_path().get_file());
+		} else {
+			fx->set_text(0, afx->get_name());
+		}
+
+		if (!afx->is_connected(SNAME("changed"), callable_mp(this, &EditorAudioBus::update_bus))) {
+			afx->connect(SNAME("changed"), callable_mp(this, &EditorAudioBus::update_bus));
+		}
```

Comment:
```suggestion
		afx->connect_changed(callable_mp(this, &EditorAudioBus::update_bus));
```
This was added recently.

Reply:
Oh great ! Will commit right away !